packages/client/src/helpers/DynascaleManager.ts

```diff
@@ -45,17 +68,151 @@ export class DynascaleManager {
+    map((overrides) => {
+      const { [globalOverrideKey]: globalSettings, ...participants } =
+        overrides;
+      return {
+        enabled: globalSettings?.enabled !== false,
+        preferredResolution: globalSettings?.enabled
+          ? globalSettings.dimension
+          : undefined,
+        participants: Object.fromEntries(
+          Object.entries(participants).map(
+            ([sessionId, participantOverride]) => [
+              sessionId,
+              {
+                enabled: participantOverride?.enabled !== false,
+                preferredResolution: participantOverride?.enabled
+                  ? participantOverride.dimension
+                  : undefined,
+              },
+            ],
+          ),
+        ),
+        isParticipantVideoEnabled: (sessionId: string) =>
+          overrides[sessionId]?.enabled ??
+          overrides[globalOverrideKey]?.enabled ??
+          true,
+      };
+    }),
+    shareReplay(1),
+  );
 
   /**
    * Creates a new DynascaleManager instance.
    *
    * @param call the call to manage.
    */
-  constructor(call: Call) {
-    this.call = call;
+  constructor(callState: CallState, speaker: SpeakerManager) {
+    this.callState = callState;
+    this.speaker = speaker;
+  }
+
+  setSfuClient(sfuClient: StreamSfuClient | undefined) {
+    this.sfuClient = sfuClient;
+  }
+
+  get trackSubscriptions() {
+    const subscriptions: TrackSubscriptionDetails[] = [];
+    for (const p of this.callState.remoteParticipants) {
+      // NOTE: audio tracks don't have to be requested explicitly
+      // as the SFU will implicitly subscribe us to all of them,
+      // once they become available.
+      if (p.videoDimension && hasVideo(p)) {
+        const override =
+          this.videoTrackSubscriptionOverrides[p.sessionId] ??
+          this.videoTrackSubscriptionOverrides[globalOverrideKey];
+
+        if (override?.enabled !== false) {
+          subscriptions.push({
+            userId: p.userId,
+            sessionId: p.sessionId,
+            trackType: TrackType.VIDEO,
+            dimension: override?.dimension ?? p.videoDimension,
+          });
+        }
+      }
+      if (p.screenShareDimension && hasScreenShare(p)) {
+        subscriptions.push({
+          userId: p.userId,
+          sessionId: p.sessionId,
+          trackType: TrackType.SCREEN_SHARE,
+          dimension: p.screenShareDimension,
+        });
+      }
+      if (hasScreenShareAudio(p)) {
+        subscriptions.push({
+          userId: p.userId,
+          sessionId: p.sessionId,
+          trackType: TrackType.SCREEN_SHARE_AUDIO,
+        });
+      }
+    }
+    return subscriptions;
   }
 
+  get videoTrackSubscriptionOverrides() {
+    return getCurrentValue(this.videoTrackSubscriptionOverrides$);
+  }
+
+  setVideoTrackSubscriptionOverrides(
+    override: VideoTrackSubscriptionOverride | undefined,
+    sessionIds?: string[],
+  ) {
```

Comment:
Can you make this an arrow function?

Reply:
Yes, I'll go through the whole class.

Reply:
https://github.com/GetStream/stream-video-js/pull/1486/commits/dbf982231d74c77933edf89d65dc7d574c9599c6